### PR TITLE
chore(oscp_crl): port OCSP and CRL features to CE repo (re4.4)

### PR DESCRIPTION
**Pull request by thalesmg**
CE part of the OCSP + CRL features

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/` dir
- [x] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)
- [x] For internal contributor: there is a jira ticket to track this change, and another jira tickt to track doc updates (if any)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

### CODE REVIEW

File: src/emqx_app.erl [hunk 1]
```diff
@@ -78,6 +80,33 @@ start_ce_modules() ->
     ok.
 -endif.
 
+set_clientid_enrichment_module() ->
```
Comment (zmstone) [inline src/emqx_app.erl:83]: export this function (in case we need to do some dirty hack).

File: src/emqx_app.erl [hunk 2]
```diff
@@ -78,6 +80,33 @@ start_ce_modules() ->
...
+        Mod ->
+            case erlang:function_exported(Mod, enrich_clientid_alias, 2) of
+                true ->
+                    persistent_term:put(clientid_enrichment_module, Mod);
+                false ->
+                    ok
+            end
+    end.
+
+set_special_auth_module() ->
```
Comment (zmstone) [inline src/emqx_app.erl:96]: export this function, (in case we need to do some dirty hot-reload).

File: priv/emqx.schema [hunk 3]
```diff
@@ -852,6 +852,14 @@ end}.
   {datatype, string}
 ]}.
 
+{mapping, "clientid_enrichment_module", "emqx.clientid_enrichment_module", [
+  {datatype, atom}
+]}.
+
+{mapping, "special_auth_module", "emqx.special_auth_module", [
+  {datatype, atom}
+]}.
```
Comment (zmstone) [inline priv/emqx.schema:861]: add some docs

File: src/emqx_app.erl [hunk 4]
```diff
@@ -78,6 +85,33 @@ start_ce_modules() ->
...
+    case emqx:get_env(special_auth_module) of
+        undefined ->
+            ok;
+        Mod ->
+            case erlang:function_exported(Mod, check_authn, 2) of
+                true ->
+                    persistent_term:put(special_auth_module, Mod),
+                    emqx:hook('client.authenticate', fun Mod:check_authn/2, []);
```
Comment (zmstone) [inline src/emqx_app.erl:109]: we need to add it with a priority,
```
try
    Mod:authn_priority()
catch
    _:_ ->
        10000 % very priority by default.
end.
```

Comment (thalesmg) [reply src/emqx_app.erl]: :heavy_check_mark:

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
oh, forgot one thing: please add change logs
i'll help to fill in the zh tr

**Review by zmstone — APPROVED** [review]

**Pull request merged**